pkg/utils/vmi.go

```diff
@@ -49,3 +62,61 @@ func (v *VmiGetter) WhoUseNad(nad *nadv1.NetworkAttachmentDefinition, nodesFilte
 
 	return afterFilter, nil
 }
+
+// Get the vmi lists who uses a group of nads,
+// note: duplicated vmis may exist when they attache to mutli nads
+func (v *VmiGetter) WhoUseNads(nads []*nadv1.NetworkAttachmentDefinition, filterFlag bool, nodesFilter mapset.Set[string]) ([]*kubevirtv1.VirtualMachineInstance, error) {
+	vmis := make([]*kubevirtv1.VirtualMachineInstance, 0)
+	for _, nad := range nads {
+		// get all vmis on this nad
+		vmisTemp, err := v.WhoUseNad(nad, filterFlag, nodesFilter)
+		if err != nil {
+			return nil, err
+		}
+		vmis = append(vmis, vmisTemp...)
+	}
+	return vmis, nil
+}
+
+// Get the vmi name list who uses the nad
+func (v *VmiGetter) VmiNamesWhoUseNad(nad *nadv1.NetworkAttachmentDefinition, filterFlag bool, nodesFilter mapset.Set[string]) ([]string, error) {
+	vmis, err := v.WhoUseNad(nad, filterFlag, nodesFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	return generateVmiNameList(vmis), nil
+}
+
+// Get the vmi name list who uses a group of nads,
+// note: duplicated names are removed
+func (v *VmiGetter) VmiNamesWhoUseNads(nads []*nadv1.NetworkAttachmentDefinition, filterFlag bool, nodesFilter mapset.Set[string]) ([]string, error) {
+	vmis, err := v.WhoUseNads(nads, filterFlag, nodesFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	cnt := len(vmis)
+
+	if cnt == 0 {
+		return nil, nil
+	}
+
+	if cnt == 1 {
+		return generateVmiNameList(vmis), nil
+	}
+
+	// use mapset to remove duplicated names
+	return mapset.NewSet[string](generateVmiNameList(vmis)...).ToSlice(), nil
```

Comment:
is there a specific reason to check `cnt == 1` ? we could just return the mapset here

```suggestion
	return mapset.NewSet[string](generateVmiNameList(vmis)...).ToSlice(), nil
```

Reply:
When there is only 1 element, surely it has no duplicated names. The `mapset.NewSet...` is a heavy object & operation. The code tries to call it when necessary.

On the other hand, I am still thinking if we should only return the first e.g. 10 names, some cluster runs 300+ VMs, expand all the names may overflow the warning window.